Demo/src/main/java/com/paypal/android/ui/paypal/PayPalFragment.kt

```diff
@@ -55,22 +72,8 @@ class PayPalFragment : Fragment() {
         val coreConfig = CoreConfig(BuildConfig.CLIENT_ID, environment = Environment.SANDBOX)
         val application = requireActivity().application
         val returnUrl = BuildConfig.APPLICATION_ID + "://paypalpay"
-
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            payPalViewModel.setPayPalClient(PayPalClient(application, coreConfig, returnUrl))
-        }
-
-        view.findViewById<View>(R.id.payPalButton).setOnClickListener {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                launchNativeCheckout()
-            }
-        }
-
-        view.findViewById<View>(R.id.payPalCreditButton).setOnClickListener {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                launchNativeCheckout()
-            }
-        }
+        paypalClient = PayPalClient(application, coreConfig, returnUrl)
+        paypalClient.listener = this
```

Comment:
im not that convinced by this. I think that we should remove as much logic as posible from `Fragments/Activities`. I think the `PayPalClient` should live on the `viewmodel` layer rather than the UI one. But it falls more on the Demo app side rather than the sdk

Reply:
I'm open to reverting to a ViewModel centric style, but my concern is the logic in ViewModels will become more complex in the Demo app, which could make it harder to maintain.

Reply:
My two cents would be we should keep anything that requires Context in the Fragment/Activity. I don't think the View Model should be aware of Android Context.